File: CPP/27.cpp
class Solution {
public:
    int removeElement(vector<int>& nums, int val) {
        if (nums.size()  == 0) return 0;
        int index = 0;
        for (int itr = 0; itr < nums.size(); ++itr){
            // Not the element -> replace the index element
            if(nums[itr] != val){
                nums[index] = nums[itr];
                index++;
            }
        }
        return index;
    }
};

File: Python/27.py
class Solution:
    def removeElement(self, nums: List[int], val: int) -> int:
        if len(nums) == 0: return 0
        index = 0
        for itr in range(len(nums)):
            if nums[itr] != val:
                nums[index] = nums[itr]
                index += 1
        return index
